src/query/service/src/pipelines/processors/transforms/transform_async_function.rs

```diff
@@ -177,8 +169,23 @@ impl TransformAsyncFunction {
                             None
                         };
 
+                        let req = GetSequenceReq {
+                            ident: SequenceIdent::new(&tenant, sequence_name),
+                        };
+                        let resp = catalog.get_sequence(req, &visibility_checker).await?;
+                        let step_size = resp.meta.step as u64;
+
+                        // Calculate batch size - take the larger of count or step_size
+                        let batch_size = to_fetch.max(step_size);
```

Comment:
https://github.com/databendlabs/databend/pull/18659/files#r2314770512

in here.